docs/architecture/adr/adr-006-in-cluster-db.md

```diff
@@ -0,0 +1,170 @@
+# ADR 006: In-cluster Database
+
+## Context
+
+We have known, since summer 2024, that Microsoft had decided to sunset Azure
+database for MariaDB. Their recommendation has been to migrate to their Flexible
+MySql database which they have provided pretty well documented. Back then we had
+already experienced database downtime, which prompted us to start a discussion
+about wether to move the databse to Azure Flexible MySql server or to an
+in-cluster database. This conversation has been running for the better part of
+6-8 months.
+
+The main points we kept debating was:
+
+- Can we operate an in-cluster database on par or better than Azure?
+- Are we able to deal with problems that might arise on the databases with out
+Azures Technical Support?
+- If we self host the databse, are we then able to optimize the performance?
+- Can we guarantee a better or on par uptime to Azure's managed database if we
+selfhost the database?
+- Would we get better support if we moved to the suggested managed database?
+- Would the performance be better on the database suggested by Azure?
+- Would downtime be minimized if we moved to the sugggested manged database?
+
+We have continually, and to a worsening degree, seen even more downtime since
+summer. We have been cut-off from having insight into the logs of the database
+due to an extremely steep price on server logs, which we tried enabling during
+the summer. This resulted in a more than doubled operations cost, which the
+operations budget in now way can bear.
```

Comment:
I think this comes across as subjective and imprecise.

Is it important to talk about all the issues with MariaDB, and how bad it was? I think all of the issues could be summarized more briefly.

Then, I think it is more important to talk about how these issues shape our view of and expectations for Microsoft-run services in general.